### DISCUSSION TITLE
Why is `AnimationPlayer.current_animation` a `String` instead of a `StringName`?

### DISCUSSION THREAD

**Discussion by feois** [godotengine/godot-proposals#7028] [General Discussions]
I'm finding it inconsistent with the fact that the rest of `AnimationPlayer` all use `StringName` except this property. There's no reason to modify `current_animation` in place so `StringName` makes more sense.

**Comment by AThousandShips** [comment]
This seems to be an error, as the property has `StringName`, this would have to wait though as it breaks compat (even technically error)

**Reply by feois** [reply]
So should I make a proposal or file it as a bug?

**Reply by AThousandShips** [reply]
Think this is good as it is, there's technically the bug that the functions are `String` but the property has `StringName`, but doesn't seem to cause any errors so it's just a matter of convenience and performance arguably, will look into it deeper

Internally `current_animation` and `assigned_animation` are stored as a `StringName`, but `autoplay` is a `String`, and yes there's inconsistency here

The same applies for `AnimatedSprite2D/3D`

Working on some changes for this, will see how it goes, and will open a PR if it works well
